File: india_compliance/public/js/sales_invoice_list.js
const erp_onl = frappe.listview_settings["Sales Invoice"].onload;
frappe.listview_settings["Sales Invoice"].onload = function (list_view) {
    if (erp_onl) {
        erp_onl(list_view);
    }
    list_view.page.add_action_item(__("Generate IRN"), function (event) {
        const docnames = list_view.get_checked_items(true);
        frappe.call({
            method: "india_compliance.cleartax_integration.API.irn.bulk_irn",
            args: {
                data: docnames
            },
            type: "POST",
            callback: function(r) {
                frappe.msgprint("IRN is being Generated!")
                list_view.refresh();
                }
            })

    });
    list_view.page.add_action_item(__("Generate EWB"), function (event) {
        const docnames = list_view.get_checked_items(true);
        frappe.call({
            method: "india_compliance.cleartax_integration.API.ewb.bulk_ewb",
            args: {
                data: docnames
            },
            type: "POST",
            callback: function(r) {
                frappe.msgprint("EWB is being Generated!")
                list_view.refresh();
                }
            })

    });

};

File: gst_india/public/js/sales_invoice_list.py
frappe.listview_settings['Sales Invoice'] = {
    onload: function(listview) {
		listview.page.add_action_item(__("Generate IRN"), function (event) {
			let selected = [];

            for (let check of event.view.cur_list.$checks) {
                selected.push(check.dataset.name);
            }
            frappe.call({
                method: "gst_india.cleartax_integration.API.irn.bulk_irn",
                args: {
                    data: selected
                },
                type: "POST",
                callback: function(r) {
                    frappe.msgprint("IRN is being Generated!")
                    listview.refresh();
                    }
                })

		});
        listview.page.add_action_item(__("Generate EWB"), function (event) {
			let selected = [];

            for (let check of event.view.cur_list.$checks) {
                selected.push(check.dataset.name);
            }
            frappe.call({
                method: "gst_india.cleartax_integration.API.ewb.bulk_ewb",
                args: {
                    data: selected
                },
                type: "POST",
                callback: function(r) {
                    frappe.msgprint("EWB is being Generated!")
                    listview.refresh();
                    }
                })

		});
    }
}